plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -910,6 +927,375 @@ public void testSettingChangeDataFeedEnabledProperty()
+        assertThat(resultWithoutTimestamp).containsOnly(
+                row("url1", "domain1", 1, null, null, "insert", 1),
+                row("url2", "domain2", 2, "starburst", null, "insert", 3),
+                row("url3", "domain3", 3, "databricks", null, "insert", 4),
+                row("url4", "domain4", 4, "tabular", 4000L, "insert", 6),
+                row("url5", "domain5", 5, "snowflake", 5000L, "insert", 7),
+                row("url6", "domain6", 6, "amazon", 6000L, "insert", 8));
+    }
+
+    @Test
+    public void testCdfOnTableWhichDoesntHaveItEnabledInitially()
+    {
+        String tableName = "test_cdf_on_table_without_it_initially_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (page_url VARCHAR, domain VARCHAR, views INTEGER)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url1', 'domain1', 1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url2', 'domain2', 2)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url3', 'domain3', 3)", 1);
+        MaterializedResult result = computeActual("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "', 0))");
+        List<MaterializedRow> resultWithoutTimestamp = getCdfColumnsWithoutTimestamp(result);
+        assertThat(resultWithoutTimestamp).containsOnly(
+                row("url1", "domain1", 1, "insert", 1),
+                row("url2", "domain2", 2, "insert", 2),
+                row("url3", "domain3", 3, "insert", 3));
+
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url22' WHERE domain = 'domain2'", 1);
+        assertQuerySucceeds("ALTER TABLE " + tableName + " SET PROPERTIES change_data_feed_enabled = true");
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url33' WHERE views = 3", 1);
+        assertUpdate("DELETE FROM " + tableName + " WHERE page_url = 'url1'", 1);
+
+        assertQuery("SELECT * FROM " + tableName, "VALUES" +
+                "('url22', 'domain2', 2), " +
+                "('url33', 'domain3', 3)");
+
+        assertQueryFails("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "', 3))",
+                "Change Data Feed is not enabled for the requested versions range");
+        assertQueryFails("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "'))",
+                "Change Data Feed is not enabled for the requested versions range");
+        result = computeActual("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "', 5))");
+        resultWithoutTimestamp = getCdfColumnsWithoutTimestamp(result);
+        assertThat(resultWithoutTimestamp).containsOnly(
+                row("url3", "domain3", 3, "update_preimage", 6),
+                row("url33", "domain3", 3, "update_postimage", 6),
+                row("url1", "domain1", 1, "delete", 7));
+    }
+
+    @Test
+    public void testReadingChangesFromCTASTableCdfEnabled()
```

Comment:
nit: per code style, ACRONYMS are written as if they were words, so "Ctas" and "Cdf"